app/models/host/managed_extensions.rb

```diff
@@ -17,7 +17,10 @@ def queue_reboot
   end
 
   def setReboot
-    old.becomes(Host::Discovered).reboot
+    old.becomes(Host::Discovered).reboot if errors.empty?
```

Comment:
Ok I see the problem, we include the discovery extension before orchestration is included itself in core:

```
[2] pry(#<Host::Managed>)> self._save_callbacks.collect(&:filter)
=> [:delete_discovery_attribute_set,
 :update_hostgroups_puppetclasses,
 "_callback_before_1088(self)",
 :autosave_associated_records_for_model,
 :autosave_associated_records_for_location,
 :autosave_associated_records_for_organization,
 :autosave_associated_records_for_search_users,
 :autosave_associated_records_for_hostgroup,
 :autosave_associated_records_for_owner,
 :autosave_associated_records_for_compute_resource,
 :autosave_associated_records_for_image,
 :clear_data_on_build,
 :autosave_associated_records_for_architecture,
 :autosave_associated_records_for_environment,
 :autosave_associated_records_for_operatingsystem,
 :autosave_associated_records_for_medium,
 :autosave_associated_records_for_ptable,
 :autosave_associated_records_for_puppet_proxy,
 :autosave_associated_records_for_puppet_ca_proxy,
 :autosave_associated_records_for_realm,
 :autosave_associated_records_for_compute_profile,
 :check_puppet_ca_proxy_is_required?,
 :crypt_root_pass,
 :before_save_collection_association,
 :on_save, # <<< THIS IS THE ORCHESTRATION MAIN LOOP
 :autosave_associated_records_for_discovery_rule]
```